### PR TITLE
CDPS-64 - Pull scheduled events from the prison API

### PR DESCRIPTION
**Pull request by danielburnley**
Pulls the data into the schedule - this change also included some minor design tweaks, the version that displays now is below

image

image

### CODE REVIEW

File: server/data/prisonApiClient.ts [hunk 1]
```diff
@@ -109,4 +111,18 @@ export default class PrisonApiRestClient implements PrisonApiClient {
       return error
     }
   }
+
+  async getEventsScheduledForToday(bookingId: number): Promise<ScheduledEvent[]> {
+    try {
+      const result = await this.restClient.get({
+        path: `/api/bookings/${bookingId}/events/today`,
+      })
+      return result as ScheduledEvent[]
```
Comment (davidmiddletonmoj) [inline server/data/prisonApiClient.ts:120]: If you type the API get call like `restClient.get<ScheduledEvent[]>()` you don't need the `as ScheduledEvent[]` on the return

File: integration_tests/e2e/overviewPage.cy.ts [hunk 2]
```diff
@@ -1,6 +1,12 @@
 import Page from '../pages/page'
 import OverviewPage from '../pages/overviewPage'
 
+const visitOverviewPage = (): OverviewPage => {
```
[the remark this answers is not in the document]
Comment (danielburnley) [reply integration_tests/e2e/overviewPage.cy.ts:4]: ah yeah I held back for conflicts at the time, thanks for the reminder 😅

File: server/data/prisonApiClient.ts (hunk 1 above)
Comment (danielburnley) [reply server/data/prisonApiClient.ts]: Did this and then also did a big refactor 🎉

**Review by davidmiddletonmoj — APPROVED** [review]
LGTM

**Pull request merged**